server/src/main/java/org/apache/druid/client/BrokerServerView.java

```diff
@@ -334,22 +408,7 @@ private void serverRemovedSegment(DruidServerMetadata server, DataSegment segmen
       }
 
       if (selector.isEmpty()) {
-        VersionedIntervalTimeline<String, ServerSelector> timeline = timelines.get(segment.getDataSource());
         selectors.remove(segmentId);
-
-        final PartitionChunk<ServerSelector> removedPartition = timeline.remove(
-            segment.getInterval(), segment.getVersion(), segment.getShardSpec().createChunk(selector)
-        );
-
-        if (removedPartition == null) {
```

Comment:
Removal of segments from timeline has now been moved to the new method `polledSegmentsFromCoordinator`. We should move this logging and callback execution there instead of completely deleting it.

Reply:
Moved. Incase of segment removal, I am removing it from the `selectors` map and triggering call back execution for each server present in the corresponding `ServerSelector`.